core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -610,10 +610,10 @@ private void readCurrentBlockToPos(long pos) throws IOException {
       return;
     }
 
-    // Do not set the buffer size too small to avoid slowing down seek by too much.
     byte[] buffer = new byte[Math.min((int) mSeekBufferSizeBytes, (int) len)];
     do {
-      int bytesRead = read(buffer);
+      // Account for the last read which might be less than mSeekBufferSizeBytes bytes.
```

Comment:
not really required for this PR. but if you feel we should do this, we could leave it as a TODO